docker/src/main/java/brooklyn/entity/container/docker/DockerContainerImpl.java

```diff
@@ -491,7 +491,7 @@ public DockerContainerLocation createLocation(Map flags) {
                     .configure(flags)
                     .configure(DynamicLocation.OWNER, this)
                     .configure("machine", container) // the underlying JcloudsLocation
-                    .configure(container.config().getBag().getAllConfig())
+                    .configure(((ConfigurationSupportInternal) container.config()).getBag().getAllConfig())
```

Comment:
Not convinced we need this cast, type returned by `ontainer.config()` should be `ConfigurationSupportInternal` see https://github.com/apache/incubator-brooklyn/blob/master/core/src/main/java/brooklyn/location/basic/AbstractLocation.java#L350